src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java

```diff
@@ -13,100 +15,165 @@
+          + "  WHERE not (data #>> '{}')::jsonb ??| array['partial_dar_code', 'partialDarCode'] "
+          + "  AND draft != true ")
+  List<DataAccessRequest> findAllDataAccessRequests();
+
+  /**
+   * Find all draft/partial DataAccessRequests, sorted descending order
+   *
+   * @return List<DataAccessRequest>
+   */
+  @SqlQuery(
+      "SELECT id, reference_id, draft, user_id, create_date, sort_date, submission_date, update_date, (data #>> '{}')::jsonb AS data FROM data_access_request "
+          + "  WHERE (data #>> '{}')::jsonb ??| array['partial_dar_code', 'partialDarCode'] "
+          + "  OR draft = true "
+          + "  ORDER BY ((data #>> '{}')::jsonb->>'sortDate')::numeric DESC")
+  List<DataAccessRequest> findAllDraftDataAccessRequests();
 
-    /**
-     * Find all draft/partial DataAccessRequests, sorted descending order
-     * @return List<DataAccessRequest>
-     */
-    @SqlQuery("SELECT id, reference_id, draft, (data #>> '{}')::jsonb AS data FROM data_access_request " +
-            "  WHERE (data #>> '{}')::jsonb ??| array['partial_dar_code', 'partialDarCode'] " +
-            "  OR draft = true " +
-            "  ORDER BY ((data #>> '{}')::jsonb->>'sortDate')::numeric DESC")
-    List<DataAccessRequest> findAllDraftDataAccessRequests();
+  /**
+   * Find all draft/partial DataAccessRequests by user id, sorted descending order
+   *
+   * @return List<DataAccessRequest>
+   */
+  @SqlQuery(
+      "SELECT id, reference_id, draft, user_id, create_date, sort_date, submission_date, update_date, (data #>> '{}')::jsonb AS data FROM data_access_request "
+          + "  WHERE ( (data #>> '{}')::jsonb ??| array['partial_dar_code', 'partialDarCode'] "
+          + "          OR draft = true ) "
+          + "  AND ((data #>> '{}')::jsonb->>'userId')::numeric = :userId "
+          + "  ORDER BY ((data #>> '{}')::jsonb->>'sortDate')::numeric DESC")
+  List<DataAccessRequest> findAllDraftsByUserId(@Bind("userId") Integer userId);
 
-    /**
-     * Find all draft/partial DataAccessRequests by user id, sorted descending order
-     * @return List<DataAccessRequest>
-     */
-    @SqlQuery("SELECT id, reference_id, draft, (data #>> '{}')::jsonb AS data FROM data_access_request " +
-            "  WHERE ( (data #>> '{}')::jsonb ??| array['partial_dar_code', 'partialDarCode'] " +
-            "          OR draft = true ) " +
-            "  AND ((data #>> '{}')::jsonb->>'userId')::numeric = :userId " +
-            "  ORDER BY ((data #>> '{}')::jsonb->>'sortDate')::numeric DESC")
-    List<DataAccessRequest> findAllDraftsByUserId(@Bind("userId") Integer userId);
+  /**
+   * Find DataAccessRequest by reference id
+   *
+   * @param referenceId String
+   * @return DataAccessRequest
+   */
+  @SqlQuery(
+      "SELECT id, reference_id, draft, user_id, create_date, sort_date, submission_date, update_date, (data #>> '{}')::jsonb AS data FROM data_access_request WHERE reference_id = :referenceId limit 1")
```

Comment:
Quick question: are reference ids not unique in the db?

Reply:
They are unique. I'm anticipating that you'll ask ... why not just use the id? They also have a bit of history. When they were mongo objects, they were created by converting the hex `_id` to a string. Since moving to postgres, I've changed that to using UUIDs. We also generate consents with UUIDs, and we call those reference ids. In an election, since an election could be on different entities, we use the reference id instead of the database id.